cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -179,23 +192,36 @@ private CompletableFuture<CruiseControlRebalanceResponse> internalRebalance(Reco
                     // send request and handle response
                     LOGGER.traceCr(reconciliation, "Response: {}, body: {}", response, response.body());
                     int statusCode = response.statusCode();
+
+                    JsonNode json = null;
+                    String userTaskID = "";
+                    String errorMessage = "";
+
+                    if (statusCode == 200 || statusCode == 201 || statusCode == 202 || statusCode == 500) {
+                        try {
+                            json = objectMapper.readTree(response.body());
+                        } catch (JsonProcessingException e) {
+                            return CompletableFuture.failedFuture(new CruiseControlRestException(
+                                    "Error for request: " + host + ":" + port + path + ". Failed to deserialize the response: " + e));
+                        }
+
+                        userTaskID = response.headers().firstValue(USER_TASK_ID_HEADER).orElse("");
+
+                        if (json.has(CC_REST_API_ERROR_KEY)) errorMessage = json.get(CC_REST_API_ERROR_KEY).asText();
+                    }
```

Comment:
I understood what you are trying to achieve with the above code. You are "centralising" where the JSON is extracted from the response for the several status codes. The drawbacks I see is that if we need to handle an additional `statusCode`, we always need to add the new code here as well (and not just later on where we have the `if` branch for handling its logic).
I was wondering if it's better factoring out a private method to contain the code which extract the JSON from the response or returning the `CruiseControlRestException` exception if any error.
Using this method in all the places where we actually have `JsonObject json = new JsonObject(response.body());` and leaving the rest of the code the same as it's currently. Would it work in your opinion?

Reply:
That is a good point. I took your suggestion made an update. Thank you Paolo.